### DISCUSSION TITLE
Performance & resource consumption

### DISCUSSION THREAD

**Discussion by cyqsimon** [lichess-org/fishnet#264] [Q&A] [answered]
Hi, I've got two servers lying around with rather low utilisation - one is a dual Epyc 7282 (32C64T) and one is an Epyc 9684 (96C192T), both of which I can spare some cores to running fishnet. I would like to first ask about some questions related to performance and resource consumption:

1. Does running fishnet for an extended period of time accumulate local data (apart from logs)? If so at approximately what rate (bytes per job)?
2. Approximately how much network traffic does fishnet produce (bytes per job)?
3. How much performance benefit will likely be derived from building with `target-cpu=native`? Let's say, compared to using the docker container.
4. Is there an easy way to benchmark the performance of the compiled binary? I would like to check whether there is performance to be gained by using GCC/Clang.

Thanks in advance.

**Comment by niklasf** [comment, marked as answer]
Hi, sure, hope the following helps:

1. The only persistent data generated and updated while running is a text file `~/.fishnet-stats` that looks like:
    ```json
   {
     "total_batches": 127,
     "total_positions": 6956,
     "total_nodes": 8683220715
    }
   ```
   Even this is optional (`--no-stats-file`).
2. It depends. To give a rough indication, I am measuring 80 Kbit/s up and 6 Kbit/s down on a typical 12 core worker.
3. None. All the heavy lifting is done by Stockfish, and the Docker container has Stockfish binaries with all relevant CPU feature combinations, of which the best is automatically selected.
4. It's probably easiest to benchmark Stockfish directly, either building from source, or taking the binaries from `/tmp/fishnet-XXX` while fishnet is running.

**Reply by niklasf** [reply]
Exploring (4) sounds like a good idea for official builds, too.

**Reply by cyqsimon** [reply]
Thanks a lot for the immediate and detailed answer. The software itself and the packaging seem very well thought out (as are all other Lichess software I've looked at). Thanks for this amazing work you do!
